test/specs/core/mergeConfig.spec.js

```diff
@@ -33,30 +33,40 @@ describe('core::mergeConfig', function() {
   it('should not inherit request options', function() {
     var localDefaults = {
       method: '__sample method__',
-      params: '__sample params__',
       data: { foo: true }
     };
     var merged = mergeConfig(localDefaults, {});
     expect(merged.method).toEqual(undefined);
-    expect(merged.params).toEqual(undefined);
     expect(merged.data).toEqual(undefined);
   });
 
-  it('should merge auth, headers, proxy with defaults', function() {
-    expect(mergeConfig({ auth: undefined }, { auth: { user: 'foo', pass: 'test' } })).toEqual({
+  it('should merge auth, headers, params, proxy with defaults', function() {
+    expect(mergeConfig({ auth: { user: 'foo' } }, { auth: { pass: 'test' } })).toEqual({
       auth: { user: 'foo', pass: 'test' }
     });
-    expect(mergeConfig({ auth: { user: 'foo', pass: 'test' } }, { auth: { pass: 'foobar' } })).toEqual({
-      auth: { user: 'foo', pass: 'foobar' }
+    expect(mergeConfig({ headers: { user: 'foo' } }, { headers: { pass: 'test' } })).toEqual({
+      headers: { user: 'foo', pass: 'test' }
+    });
+    expect(mergeConfig({ params: { user: 'foo' } }, { params: { pass: 'test' } })).toEqual({
+      params: { user: 'foo', pass: 'test' }
+    });
+    expect(mergeConfig({ proxy: { user: 'foo' } }, { proxy: { pass: 'test' } })).toEqual({
+      proxy: { user: 'foo', pass: 'test' }
     });
   });
 
-  it('should overwrite auth, headers, proxy with a non-object value', function() {
+  it('should overwrite auth, headers, params, proxy with a non-object value', function() {
     expect(mergeConfig({ auth: { user: 'foo', pass: 'test' } }, { auth: false })).toEqual({
       auth: false
     });
-    expect(mergeConfig({ auth: { user: 'foo', pass: 'test' } }, { auth: null })).toEqual({
-      auth: null
```

Comment:
@textbook Would you mind reverting the removed test cases? Or is there special reason to remove them? I can understand that you want to refactor cases similar with each other, but it is better to keep them.

Reply:
Sorry, I'd somehow misread that as an exact duplicate. I've restored it to check four keys with two values each.